ocs_ci/deployment/provider_client/storage_client_deployment.py

```diff
@@ -0,0 +1,550 @@
+from ocs_ci.ocs.resources.storage_cluster import verify_storage_cluster
+from ocs_ci.ocs.resources.storage_client import create_storage_client
+from ocs_ci.ocs.resources.catalog_source import CatalogSource
+from ocs_ci.ocs.bucket_utils import check_pv_backingstore_type
+from ocs_ci.ocs.resources import pod
+from ocs_ci.helpers.helpers import (
+    get_all_storageclass_names,
+    verify_block_pool_exists,
+)
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.helpers.managed_services import (
+    verify_storageclient,
+)
+
+log = logging.getLogger(__name__)
+
+
+class ODFAndNativeStorageClientDeploymentOnProvider(object):
+    def __init__(self):
+        log.info("Initializing webdriver and login to webconsole")
+        # Call a function during initialization
+        self.initial_function()
+
+        # Register a function to be called upon the destruction of the instance
+        atexit.register(self.cleanup_function)
+
+    def initial_function(self):
+        log.info("initial_function called during initialization.")
+        login_ui()
+
+    def cleanup_function(self):
+        log.info("cleanup_function called at exit.")
+        # Remove debug namespace
+        self.ns_obj.delete_project(project_name=constants.BM_DEBUG_NODE_NS)
+        # Close browser
+        close_browser()
+
+    @pytest.fixture(scope="class", autouse=True)
+    def setup(self):
+        """
+        Setup method for the class
+
+        """
+        self.validation_ui_obj = ValidationUI()
+        self.ns_obj = ocp.OCP(kind=constants.NAMESPACES)
+        self.ns_obj.new_project(
+            project_name=constants.BM_DEBUG_NODE_NS, policy=constants.PSA_PRIVILEGED
+        )
+        self.ocp_obj = ocp.OCP()
+        self.storage_cluster_obj = ocp.OCP(
+            kind="Storagecluster", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        self.storage_profile_obj = ocp.OCP(
+            kind="Storageprofile", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        self.config_map_obj = ocp.OCP(
+            kind="Configmap", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        self.pod_obj = ocp.OCP(
+            kind="Pod", namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        self.scheduler_obj = ocp.OCP(
+            kind=constants.SCHEDULERS_CONFIG,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+        self.sc_obj = ocp.OCP(kind=constants.STORAGECLASS)
+        self.storageclass = "localblock"
+        self.ocp_version = version.get_semantic_ocp_version_from_config()
+        self.ocs_version = version.get_semantic_ocs_version_from_config()
+        self.storage_class_claims = [
+            constants.CEPHBLOCKPOOL_SC,
+            constants.CEPHFILESYSTEM_SC,
+        ]
+        self.ocs_client_operator = defaults.OCS_CLIENT_OPERATOR_NAME
+        self.deployment = Deployment()
+        self.rados_utils = RadosHelper()
```

Comment:
RadosHelper requires positional argument `mon`. Without it the code fails:
2024-05-30 13:21:58          self.ocs_client_operator = defaults.OCS_CLIENT_OPERATOR_NAME
2024-05-30 13:21:58          self.deployment = Deployment()
2024-05-30 13:21:58  >       self.rados_utils = RadosHelper()
2024-05-30 13:21:58  E       TypeError: __init__() missing 1 required positional argument: 'mon'
2024-05-30 13:21:58  
2024-05-30 13:21:58  ocs_ci/deployment/provider_client/storage_client_deployment.py:94: TypeError

Reply:
Addressed in latest commit.